vars/signArtifacts.groovy

```diff
@@ -0,0 +1,33 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ */
+
+void call(Map args = [:]) {
+
+    environment {
+        // These ENV variables are required by https://github.com/opensearch-project/opensearch-signer-client
+        // This client is invoked internally by the sign script.
+        ROLE = "${SIGNER_CLIENT_ROLE}"
+        EXTERNAL_ID = "${SIGNER_CLIENT_EXTERNAL_ID}"
+        UNSIGNED_BUCKET = "${SIGNER_CLIENT_UNSIGNED_BUCKET}"
+        SIGNED_BUCKET = "${SIGNER_CLIENT_SIGNED_BUCKET}"
+    }
+
+    if( !fileExists("$WORKSPACE/sign.sh")) {
+        git url: 'https://github.com/opensearch-project/opensearch-build.git', branch: 'main'
+    }
+
+    if( !fileExists("$WORKSPACE/opensearch.pgp")) {
+        sh('curl https://artifacts.opensearch.org/publickeys/opensearch.pgp -o $WORKSPACE/opensearch.pgp')
```

Comment:
Use `-SL` so that if the original link is redirected, it will still find the file.